cupyx/texture/_affine.py

```diff
@@ -0,0 +1,159 @@
+import cupy
+
+from cupy.cuda import texture
+from cupy.cuda import runtime
+
+
+def _create_texture_object(data,
+                           address_mode: str,
+                           filter_mode: str,
+                           read_mode: str):
+
+    if cupy.issubdtype(data.dtype, cupy.unsignedinteger):
+        fmt_kind = runtime.cudaChannelFormatKindUnsigned
+    elif cupy.issubdtype(data.dtype, cupy.integer):
+        fmt_kind = runtime.cudaChannelFormatKindSigned
+    elif cupy.issubdtype(data.dtype, cupy.floating):
+        fmt_kind = runtime.cudaChannelFormatKindFloat
+    else:
+        raise ValueError('Unsupported data type')
+
+    if address_mode == 'wrap':
+        address_mode = runtime.cudaAddressModeWrap
+    elif address_mode == 'clamp':
+        address_mode = runtime.cudaAddressModeClamp
+    elif address_mode == 'mirror':
+        address_mode = runtime.cudaAddressModeMirror
+    elif address_mode == 'border':
+        address_mode = runtime.cudaAddressModeBorder
+    else:
+        raise ValueError(
+            'Unsupported address mode '
+            '(supported: wrap, clamp, mirror, border)')
+
+    if filter_mode == 'nearest':
+        filter_mode = runtime.cudaFilterModePoint
+    elif filter_mode == 'linear':
+        filter_mode = runtime.cudaFilterModeLinear
+    else:
+        raise ValueError(
+            'Unsupported filter mode (supported: nearest, linear)')
+
+    if read_mode == 'element_type':
+        read_mode = runtime.cudaReadModeElementType
+    elif read_mode == 'normalized_float':
+        read_mode = runtime.cudaReadModeNormalizedFloat
+    else:
+        raise ValueError(
+            'Unsupported read mode '
+            '(supported: element_type, normalized_float)')
+
+    texture_fmt = texture.ChannelFormatDescriptor(
+        data.itemsize * 8, 0, 0, 0, fmt_kind)
```

Comment:
Is there a way for us to explore the multi-channel support? Just curious.

Reply:
It looks like that is already in the TODO list up top

Reply:
Ah thanks Greg. I must have missed that...